mlflow/genai/judges/custom_prompt_judge.py

```diff
@@ -0,0 +1,155 @@
+import re
+from typing import Callable, Optional, Union
+
+from mlflow.entities.assessment import Feedback
+from mlflow.entities.assessment_source import AssessmentSource, AssessmentSourceType
+from mlflow.genai.judges.builtin import _MODEL_API_DOC
+from mlflow.genai.judges.utils import (
+    format_prompt,
+    get_default_model,
+    invoke_judge_model,
+)
+from mlflow.utils.annotations import experimental
+from mlflow.utils.docstring_utils import format_docstring
+
+_CHOICE_PATTERN = re.compile(r"\[\[([\w ]+)\]\]")
+
+
+@format_docstring(_MODEL_API_DOC)
+@experimental(version="3.0.0")
+def custom_prompt_judge(
+    *,
+    name: str,
+    prompt_template: str,
+    numeric_values: Optional[dict[str, Union[int, float]]] = None,
```

Comment:
```suggestion
    numeric_values: Optional[dict[str, float]] = None,
```

let's use `float` here. Ref: https://stackoverflow.com/a/50928627

Reply:
oh didn't know that!